### ISSUE TITLE
Error responses seem to always be in text/plain

### ISSUE THREAD

**Issue by petervo** [kubernetes/kubernetes#52754]
/kind bug

**What happened**: Error data seems to always be text/plain

**What you expected to happen**: I expect to get json data. I would expect it to be by default, but especially should happen when the Accept header is set to application/json.

**How to reproduce it (as minimally and precisely as possible)**:

```
curl -vv -k -X POST -H "Content-Type: application/json" -H "Accept: application/json, */*" -d '{"apiVersion":"v1","kind":"Namespace","metadata":{"name":"test"}}' https://127.0.0.1:8443/api/v1/namespaces

> POST /api/v1/namespaces HTTP/2
> Host: 127.0.0.1:8443
> User-Agent: curl/7.53.1
> Content-Type: application/json
> Accept: application/json, */*
> Content-Length: 65
> 

< HTTP/2 403 
< cache-control: no-store
< content-type: text/plain
< x-content-type-options: nosniff
< content-length: 140
< date: Tue, 19 Sep 2017 23:19:08 GMT
< 
User "system:anonymous" cannot create namespaces at the cluster scope: User "system:anonymous" cannot create namespaces at the cluster scope
```

**Environment**:
- Kubernetes version (use `kubectl version`):

Tested with:

```
  "major": "1",
  "minor": "7",
  "gitVersion": "v1.7.5",
  "gitCommit": "17d7182a7ccbb167074be7a87f0a68bd00d58d97",
  "gitTreeState": "clean",
  "buildDate": "2017-08-31T08:56:23Z",
  "goVersion": "go1.8.3",
  "compiler": "gc",
  "platform": "linux/amd64"
```
and 

```
  "major": "1",
  "minor": "7",
  "gitVersion": "v1.7.0+80709908fd",
  "gitCommit": "930b5c4",
  "gitTreeState": "clean",
  "buildDate": "2017-09-19T21:20:25Z",
  "goVersion": "go1.8.3",
  "compiler": "gc",
  "platform": "linux/amd64"
```

- Cloud provider or hardware configuration**: Bare metal, GCE
- OS (e.g. from /etc/os-release): Fedora 26
- Kernel (e.g. `uname -a`): Linux localhost.localdomain 4.12.11-300.fc26.x86_64 #1 SMP Thu Sep 7 18:32:12 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Comment by mbohlool** [comment]
cc @jpbetz

**Comment by hzxuzhonghu** [comment]
I think its fixed in 1.9
```
root@SZX1000353068:~# kubectl version
Client Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.0-alpha.2.2994+3cf7000b3b5947", GitCommit:"3cf7000b3b5947e957f1f54faa75680bc51a987c", GitTreeState:"clean", BuildDate:"2017-09-02T08:52:35Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9+", GitVersion:"v1.9.0-alpha.0.722+cb36547ec7094a-dirty", GitCommit:"cb36547ec7094a1e496e7208028ebfd4b490e2b2", GitTreeState:"dirty", BuildDate:"2017-09-22T02:40:27Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

root@SZX1000353068:~# curl -vv -k -X POST -H "Content-Type: application/json" -H "Accept: application/json, */*" -d '{"apiVersion":"v1","kind":"Namespace","metadata":{"name":"test"}}' https://127.0.0.1:6443/api/v1/namespaces
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 6443 (#0)
* successfully set certificate verify locations:
*   CAfile: none
  CApath: /etc/ssl/certs
* SSLv3, TLS handshake, Client hello (1):
* SSLv3, TLS handshake, Server hello (2):
* SSLv3, TLS handshake, CERT (11):
* SSLv3, TLS handshake, Server key exchange (12):
* SSLv3, TLS handshake, Request CERT (13):
* SSLv3, TLS handshake, Server finished (14):
* SSLv3, TLS handshake, CERT (11):
* SSLv3, TLS handshake, Client key exchange (16):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSL connection using ECDHE-RSA-AES128-GCM-SHA256
* Server certificate:
* 	 subject: CN=kubernetes.default
* 	 start date: 2017-09-22 02:36:00 GMT
* 	 expire date: 2022-09-21 02:36:00 GMT
* 	 issuer: C=xx; ST=x; L=x; O=x; OU=x; CN=ca; emailAddress=x
* 	 SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
> POST /api/v1/namespaces HTTP/1.1
> User-Agent: curl/7.35.0
> Host: 127.0.0.1:6443
> Content-Type: application/json
> Accept: application/json, */*
> Content-Length: 65
>
* upload completely sent off: 65 out of 65 bytes
< HTTP/1.1 403 Forbidden
< Audit-Id: dcb08dc0-0925-4a24-81a6-2e3ccfb0ce1b
< Content-Type: application/json
< X-Content-Type-Options: nosniff
< Date: Fri, 22 Sep 2017 06:14:17 GMT
< Content-Length: 291
<
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {

  },
  "status": "Failure",
  "message": "namespaces is forbidden: User \"system:anonymous\" cannot create namespaces at the cluster scope",
  "reason": "Forbidden",
  "details": {
    "kind": "namespaces"
  },
  "code": 403
* Connection #0 to host 127.0.0.1 left intact
}
```

**Comment by liggitt** [comment]
fixed in 1.8
